src/journal/journald-server.c

```diff
@@ -139,9 +118,46 @@ static int determine_space_for(
                 if (!S_ISREG(st.st_mode))
                         continue;
 
-                sum += (uint64_t) st.st_blocks * 512UL;
+                *used += (uint64_t) st.st_blocks * 512UL;
+        }
+
+        return 0;
+}
+
+static int determine_space_for(
+                Server *s,
+                JournalMetrics *metrics,
+                const char *path,
+                const char *name,
+                bool verbose,
+                bool patch_min_use,
+                uint64_t *available,
+                uint64_t *limit) {
+
+        uint64_t sum, avail, ss_avail;
+        _cleanup_closedir_ DIR *d = NULL;
+        usec_t ts;
+
+        assert(s);
+        assert(metrics);
+        assert(path);
+        assert(name);
+
+        ts = now(CLOCK_MONOTONIC);
+
+        if (!verbose && s->cached_space_timestamp + RECHECK_SPACE_USEC > ts) {
+
+                if (available)
+                        *available = s->cached_space_available;
+                if (limit)
+                        *limit = s->cached_space_limit;
+
+                return 0;
         }
 
+        if (determine_path_usage(s, path, &sum, &ss_avail) < 0)
+                return -errno;
```

Comment:
error check is wrong. should be `r = determine_…; if (r < 0) return r;`. In systemd we always pass errors as negative return values, and we should do so here, too.